extras/gizmo/tri.js

```diff
@@ -0,0 +1,74 @@
+import {
+    Vec3
+} from 'playcanvas';
+
+// temporary variables
+const e1 = new Vec3();
+const e2 = new Vec3();
+const h = new Vec3();
+const s = new Vec3();
+const q = new Vec3();
+
+// constants
+const EPSILON = 1e-6;
+
+class Tri {
+    v0;
+
+    v1;
+
+    v2;
+
+    constructor(v0 = new Vec3(), v1 = new Vec3(), v2 = new Vec3()) {
+        if (v0.length === 3) {
+            this.v0 = v0[0];
+            this.v1 = v0[1];
+            this.v2 = v0[2];
+        } else {
+            this.v0 = v0;
```

Comment:
we should not keep references to user allocated Vec3's but need to copy data to our own Vec3's. For example see how Plane class does it in the engine.